product_logistics_uom/models/product_template.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2020 Akretion (https://www.akretion.com).
+# @author Raphaël Reverdy <raphael.reverdy@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    volume = fields.Float(
+        help="The volume of the contents, not including any packaging, etc."
```

Comment:
I thinkg you should also change the text for the ``product.product`` field, don't you think ? 

I mean, overloading this value https://github.com/odoo/odoo/blob/12.0/addons/product/models/product.py#L145

Reply:
> However, If you change the meaning of the volume and the weight fields. (ie = product A : `volume = 1` means 1 liter ; product B : `volume = 1` means 1 mililiter), I guess it will be incompatible with all the other OCA modules.

Yes, it's not that simple. Some points:

1) in core v12, there is a weight_uom_id. Most of the modules don't take it in account, and it's an issue by itself. Because you can choose to switch to imperial units and these modules will provide wrong results.
2) the volume is hardcoded to m³ in v12 but not in v13. (in v13 the switch to different system will still be a problem).
3) some modules just don't care about the unit, **since it's always the same**. A sum of weight if they are both in the same unit is straightforward.

So, some other OCA modules may be patched to use weight_uom_id and be compliant with this module even without any dependency.
It will not work for volume.

But the thing is that in v13 actually, weight_uom_id has been removed.
I have suggested to odoo/odoo this PR https://github.com/odoo/odoo/pull/50252 yesterday and I hope _uom_id to be back as first class citizen but we do not know how it will end.

> So, what is the road map ? Making all the OCA modules depends on your new module ? or mark your module as incompatible with all the OCA modules that are using the two fields (`weight`, `volume`) ?

So for the future,  I see different paths:
1) odoo accept my PR, so OCA modules can be updated to depend of *_uom_id in their calculation. It will be little work for everyone but no change in dependencies. This module will be still relevant for the use case where you need different uom between products.
2) odoo reject my PR,  but within OCA we agree on the use of *_uom_id, so this module becomes a dependency. Little work for everyone.
3) no adhesion of the use of _uom_id, there can be some incompatibilities between oca modules.


To recap, this module does 3 things:
1) add [product.template]volume_uom_id  to be in pair with [product.template]weight_uom_id
2) allow to change default  weight_uom_id and volume_uom_id
3) allow to change product per product the right *_uom_id 

Note: in v12 there is no m³ in uom/data only liters.